packages/docs-infra/src/CodeHighlighter/CodeHighlighterClient.tsx

```diff
@@ -14,6 +14,9 @@ import * as Errors from './errors';
 
 const DEBUG = false; // Set to true for debugging purposes
 
+const requestIdleCallback = window.requestIdleCallback ?? setTimeout;
```

Comment:
I saw this being used in https://github.com/mui/base-ui/blob/v1.0.0-beta.4/docs/src/components/QuickNav/QuickNav.tsx#L217-L218